drivers/gpio/gpio_infineon_cat1.c

```diff
@@ -0,0 +1,332 @@
+		 * is not supported. Refer to SWINTEGRATION-696
+		 */
+		/* event = CYHAL_GPIO_IRQ_BOTH; */
+		return -ENOTSUP;
+
+	default:
+		return -ENOTSUP;
+	}
+
+	cyhal_gpio_t gpio_pin = CYHAL_GET_GPIO(GET_PORT_NUM(dev), pin);
+	cyhal_gpio_callback_data_t *cb_data_ptr = cfg->cb_data_ptr;
+
+	/* Find index of free callback data structure */
+	uint32_t index;
+
+	for (index = 0u; index < cfg->ngpios; index++) {
+		if ((cb_data_ptr[index].callback == NULL) || (cb_data_ptr[index].pin == gpio_pin)) {
+			break;
+		}
+	}
+
+	if (index != cfg->ngpios) {
+		/* Store callback data: gpio callback and gpio device driver handle */
+		cb_data_ptr[index].callback = &gpio_event_callback;
+		cb_data_ptr[index].callback_arg = (void *)(gpio_pin);
+
+		/* Register/clear a callback handler for pin events */
+		cyhal_gpio_register_callback(gpio_pin, (mode == GPIO_INT_MODE_DISABLED) ?
+					     NULL : &cb_data_ptr[index]);
+
+		/* Enable/disable the specified GPIO event */
+		cyhal_gpio_enable_event(gpio_pin, event, cfg->intr_priority,
+					(mode == GPIO_INT_MODE_DISABLED) ? false : true);
+		return 0;
+	} else {
+		return -EINVAL;
+	}
+}
+
+static int gpio_cat1_manage_callback(const struct device *port,
+				     struct gpio_callback *callback,
+				     bool set)
+{
+	return gpio_manage_callback(&DEV_DATA(port)->callbacks, callback, set);
+}
+
+static const struct gpio_driver_api gpio_cat1_api = {
+	.pin_configure = gpio_cat1_configure,
+	.port_get_raw = gpio_cat1_port_get_raw,
+	.port_set_masked_raw = gpio_cat1_port_set_masked_raw,
+	.port_set_bits_raw = gpio_cat1_port_set_bits_raw,
+	.port_clear_bits_raw = gpio_cat1_port_clear_bits_raw,
+	.port_toggle_bits = gpio_cat1_port_toggle_bits,
+	.pin_interrupt_configure = gpio_cat1_pin_interrupt_configure,
+	.manage_callback = gpio_cat1_manage_callback,
+	.get_pending_int = gpio_cat1_get_pending_int,
+};
+
+static int gpio_cat1_init(const struct device *dev)
+{
+	ARG_UNUSED(dev);
+	return 0;
+}
+
+#define GPIO_CAT1_INIT(n)							     \
+										     \
+	cyhal_gpio_callback_data_t						     \
+	_cat1_gpio##n##_cb_data[DT_INST_PROP(n, ngpios)];			     \
+										     \
+	static const struct gpio_cat1_config _cat1_gpio##n##_config = {		     \
+		.common = {							     \
+				.port_pin_mask = GPIO_PORT_PIN_MASK_FROM_DT_INST(n), \
+			},							     \
+		.intr_priority = DT_INST_IRQ_BY_IDX(n, 0, priority),		     \
+		.cb_data_ptr = _cat1_gpio##n##_cb_data,				     \
+		.ngpios = DT_INST_PROP(n, ngpios),				     \
+		.regs = (GPIO_PRT_Type *)DT_INST_REG_ADDR(n),			     \
+	};									     \
+										     \
+	static struct gpio_cat1_data _cat1_gpio##n##_data = { 0 };		     \
+										     \
+	DEVICE_DT_INST_DEFINE(n, gpio_cat1_init, NULL,				     \
+			      &_cat1_gpio##n##_data,				     \
+			      &_cat1_gpio##n##_config, POST_KERNEL,		     \
+			      CONFIG_KERNEL_INIT_PRIORITY_DEVICE,		     \
+			      &gpio_cat1_api);
+
+DT_INST_FOREACH_STATUS_OKAY(GPIO_CAT1_INIT)
```

Comment:
can you confirm GPIO tests pass for this driver?

Reply:
we run those GPIO tests:
INFO    - 264/266 cy8cproto_062_4343w       tests/drivers/gpio/gpio_basic_api/drivers.gpio.2pin PASSED (device 76.524s)
INFO    - 265/266 cy8cproto_062_4343w       tests/drivers/gpio/gpio_api_1pin/peripheral.gpio.1pin PASSED (device 74.724s)
[peripheral.gpio.1pin_handler.log](https://github.com/zephyrproject-rtos/zephyr/files/8889072/peripheral.gpio.1pin_handler.log)
[drivers.gpio.2pin_handler.log](https://github.com/zephyrproject-rtos/zephyr/files/8889073/drivers.gpio.2pin_handler.log)